kalite/distributed/management/commands/screenshots.py

```diff
@@ -0,0 +1,338 @@
+        else:
+            self.output_path = settings.SCREENSHOTS_OUTPUT_PATH
+        ensure_dir(self.output_path)
+
+        # make sure directory is empty from screenshot files
+        png_path = os.path.join(self.output_path, "*%s" % settings.SCREENSHOTS_EXTENSION)
+        pngs = glob.glob(png_path)
+        if pngs and not kwargs['no_del']:
+            self.logwarn("==> Deleting existing screenshots: %s ..." % png_path)
+            for filename in pngs:
+                os.remove(filename)
+
+        # setup database to use and auto-create admin user
+        self.loginfo("==> Setting-up database ...")
+        self.admin_user = reset_sqlite_database(self.admin_username, self.admin_email, self.default_password, verbosity=self.verbosity)
+        self.admin_pass = self.default_password
+        if not self.admin_user:
+            raise Exception("==> Did not successfully setup database!")
+
+        self.facility = self.create_facility()
+        self.create_student(username=self.student_username, password=self.default_password)
+        self.create_teacher(username=self.coach_username, password=self.default_password)
+
+        self.persistent_browser = True
+        self.max_wait_time = kwargs.get('max_wait_time', 30)
+
+        self.setUpClass()
+
+        self.loginfo("==> Setting-up browser ...")
+        super(Screenshot, self).setUp()
+        self.browser.set_window_size(1024, 768)
+
+        self.loginfo("==> Browser %s successfully setup with live_server_url %s." %
+                 (self.browser.name, self.live_server_url,))
+        self.loginfo("==> Saving screenshots to %s ..." % (settings.SCREENSHOTS_OUTPUT_PATH,))
+
+    def validate_json_keys(self, shot):
+        """
+        Validates a json item if keys are valid or not, raises an exception if keys are found.
+        """
+        values = [shot[key] for key in self.JSON_KEYS]
+
+    def make_filename(self, slug):
+        filename = "%s/%s%s" % (self.output_path, slug, settings.SCREENSHOTS_EXTENSION)
+        return filename
+
+    def snap(self, slug, focus, note):
+        filename = self.make_filename(slug=slug)
+        self.loginfo('====> Snapping %s --titled-- "%s" --> %s%s ...' %
+                 (self.browser.current_url, self.browser.title, slug, settings.SCREENSHOTS_EXTENSION))
+
+        if focus:
+            # Apply the specified styles to element. Currently only selection by
+            # id is supported. TODO: Extend it a more generic CSS selector.
+            selector = focus['selector']
+            styles = focus['styles']
+            for key, value in styles.iteritems():
+                self.browser.execute_script('$("%s").css("%s", "%s");' % (selector, key, value))
+            if note:
+                # Assuming we've loaded jquery
+                # Positioned at the bottom left, so might cause issues with some elements?
+                self.browser.execute_script("$('body').append(\"<span id='annotation'></span>\");" \
```

Comment:
Indeed, the trailing `\` are optional here. I included them to make the intention more explicit, but if you think its un-idiomatic I'll remove it.